Objects/bytearrayobject.c

```diff
@@ -1337,19 +1356,19 @@ bytearray_rindex_impl(PyByteArrayObject *self, PyObject *sub,
                       Py_ssize_t start, Py_ssize_t end)
 /*[clinic end generated code: output=38e1cf66bafb08b9 input=7d198b3d6b0a62ce]*/
 {
-    return _Py_bytes_rindex(PyByteArray_AS_STRING(self), PyByteArray_GET_SIZE(self),
-                            sub, start, end);
+    return _bytearray_with_buffer(self, sub, start, end, _Py_bytes_rindex);
 }
 
 static int
 bytearray_contains(PyObject *self, PyObject *arg)
 {
     int ret;
-    Py_BEGIN_CRITICAL_SECTION(self);
```

Comment:
I think this critical section is still needed; `ob_exports` which gets incremented when exporting the buffer prevents resizing the array but does not prevent changing the contents; the critical section here guarantees during the sequence contains (https://docs.python.org/3/c-api/typeobj.html#c.PySequenceMethods.sq_contains) the bytearray isn't modified by other python calls.